From 0fecc832fd8e72b66aaa9a53ea5d20cd3aae4c6b Mon Sep 17 00:00:00 2001
From: Peter Yin <peter.yin@quantatw.com>
Date: Thu, 23 Jan 2025 14:27:45 +0800
Subject: [PATCH 07/16] psusensors: fixed multiple power state issue

Link:https://gerrit.openbmc.org/c/openbmc/dbus-sensors/+/77391

If the entity does not set powerState for each channel,
label_powerState will be contaminated by the state of the previous
channel. Therefore, if label_powerState is not found,the powerState
should be updated.

Change-Id: I61d68834a006dec9f355d1b3f2a339ee30ada89d
Signed-off-by: Peter Yin <peter.yin@quantatw.com>
---
 src/psu/PSUSensorMain.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/psu/PSUSensorMain.cpp b/src/psu/PSUSensorMain.cpp
index 7fe1c89..1978f75 100644
--- a/src/psu/PSUSensorMain.cpp
+++ b/src/psu/PSUSensorMain.cpp
@@ -845,6 +845,10 @@ static void createSensorsCallback(
                                                     findPowerState->second);
                 setReadState(powerState, readState);
             }
+            else
+            {
+                readState = getPowerState(*baseConfig);
+            }
             if (!(psuProperty.minReading < psuProperty.maxReading))
             {
                 lg2::error("Min must be less than Max");
-- 
2.47.3

